File: DitoSDKSample/app/src/main/java/ioasys/com/br/sampleapp/MainActivity.java
package ioasys.com.br.sampleapp;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import ioasys.com.br.sacapp.SACappSDK;
import ioasys.com.br.sacapp.app.SACappSDKCallback;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onClick(View view) {

        try {
            SACappSDK.startProcess(this, "json", new SACappSDKCallback() {
                @Override
                public void onSucess(String response) {

                }
                @Override
                public void onError(Exception ex) {

                }
            });
        } catch (Exception ex) {

        }
    }
}

File: app/src/main/java/com/example/flutter_sdk_test/MainActivity.kt
package com.example.flutter_sdk_test

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.runtime.Composable
import androidx.compose.ui.tooling.preview.Preview
import br.com.dito.ditosdk.CustomData
import br.com.dito.ditosdk.Dito
import br.com.dito.ditosdk.Event
import br.com.dito.ditosdk.Identify
import com.google.android.gms.tasks.OnCompleteListener
import com.google.firebase.messaging.FirebaseMessaging

class MainActivity : AppCompatActivity() {

    private var token: String? = null
    private val pushNotificationPermissionLauncher
        get() = registerForActivityResult(ActivityResultContracts.RequestPermission()) {}

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        pushNotificationPermissionLauncher.launch(android.Manifest.permission.POST_NOTIFICATIONS)
    }

    fun onClickIdentify(view: View) {
        val data = CustomData()
        data.add("userId", "uol")
        data.add("push", false)

        val identify = Identify("44444444444")
        identify.data = data
        identify.name = "Usuário de teste Android"
        identify.email = "teste+android@dito.com.br"

        Dito.identify(identify) { registerToken() }
    }

    private fun registerToken() {
        FirebaseMessaging.getInstance().token.addOnCompleteListener(OnCompleteListener { task ->
            if (!task.isSuccessful) {
                Log.w("Dito", "Fetching FCM registration token failed", task.exception)
                return@OnCompleteListener
            }
            token = task.result
            Dito.registerDevice(token!!)
            Log.d("identify", "success on registering token")
        })
    }

    fun onClickEvent(view: View) {
        Dito.track(Event("comprou", 2.5))
    }

    fun onClickRegisterToken(view: View) {
        registerToken()
    }

    fun onClickDeleteToken(view: View) {
        token?.let {
            Dito.unregisterDevice(it)
        }
    }
}

@Preview(apiLevel = 33)
@Composable
fun AppPreview() {

}